channels/http.py

```diff
@@ -249,6 +249,29 @@ def handle_uncaught_exception(self, request, resolver, exc_info):
                 content_type="text/plain",
             )
 
+    def load_middleware(self):
+        """
+        Loads the Django middleware chain and caches it on the class.
+        """
+        # Because we create an AsgiHandler on every HTTP request
+        # we need to preserve the Django middleware chain once we load it.
+        if self.__class__._middleware_chain:
+            self._middleware_chain = self.__class__._middleware_chain
+            self._view_middleware = self.__class__._view_middleware
+            self._template_response_middleware = (
+                self.__class__._template_response_middleware
+            )
+            self._exception_middleware = self.__class__._exception_middleware
```

Comment:
Is this really necessary?

Reply:
I fear it is. 
Doing the following instead:
```python
    def load_middleware(self):
        """
        Loads the Django middleware chain and caches it on the class.
        """
        # Because we create an AsgiHandler on every HTTP request
        # we need to preserve the Django middleware chain once we load it.
        if (
            not hasattr(self.__class__, "_middleware_chain")
            or self.__class__._middleware_chain is None
        ):
            super(AsgiHandler, self).load_middleware()
            self.__class__._middleware_chain = self._middleware_chain
            self.__class__._view_middleware = self._view_middleware
            self.__class__._template_response_middleware = (
                self._template_response_middleware
            )
            self.__class__._exception_middleware = self._exception_middleware
```
yields this error:
```
ERROR    Exception inside application: inner() takes 1 positional argument but 2 were given
   File "/usr/local/lib/python3.6/site-packages/channels/http.py", line 195, in __call__
     await self.handle(body)
   File "/usr/local/lib/python3.6/site-packages/asgiref/sync.py", line 108, in __call__
     return await asyncio.wait_for(future, timeout=None)
   File "/usr/local/lib/python3.6/asyncio/tasks.py", line 339, in wait_for
     return (yield from fut)
   File "/usr/local/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
   File "/usr/local/lib/python3.6/site-packages/asgiref/sync.py", line 123, in thread_handler
     return self.func(*args, **kwargs)
   File "/usr/local/lib/python3.6/site-packages/channels/http.py", line 228, in handle
     response = self.get_response(request)
   File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 81, in get_response
     response = self._middleware_chain(request)
   inner() takes 1 positional argument but 2 were given
```

Reply:
Ah, I see, at least `_middleware_chain` is a method.